### PR TITLE
E6V2 Bootmapper Client QMK port

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description

Port the E6V2 Bootmapper Client PCB to QMK. 
Included bootmagic lite functionality set to the Esc key located at K50. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  QMKBEST = SAFE_RANGE,
+  QMKURL
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_60_ansi(
+        KC_GESC, KC_1,  KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,    KC_9,   KC_0,    KC_MINS, KC_EQL, KC_BSPC,
+        EEP_RST,  KC_Q,  KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,    KC_O,   KC_P,    KC_LBRC, KC_RBRC,KC_BSLS,
```
Comment (noroadsleft) [inline keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c]: Are `EEP_RST`, `RGB_TOG` and `RGB_MOD` supposed to be on the base layer?

https://github.com/qmk/qmk_firmware/blob/0152bbcc68f7d95a2aebffe0f7f9495f5ac9318c/keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c#L27-L29

Comment (mechmerlin) [reply keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c]: This is actually intentional. I realized that people will most likely want to test the pcb out before they solder on switches and it's incredibly difficult to reset the board or hold Fn keys without actual switches. What are your thoughts.

Comment (drashna) [reply keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c]: use TG for a function key, instead?

File: keyboards/exclusive/e6v2/bmc/rules.mk [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+RGBLIGHT_CUSTOM_DRIVER = yes
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+
+SRC += i2c.c
```
Comment (drashna) [inline keyboards/exclusive/e6v2/bmc/rules.mk]: cries in i2c

File: keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c (hunk 1 above)
Comment (noroadsleft) [reply keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c]: I feel very mixed about this.

You're basically requiring the user to write, compile and flash their own keymap after testing is done, though you could certainly make the argument that someone who buys this PCB would know how to do this (an E6-V2 is not a starter board for someone new to mechanical keyboards, IMO). I'm not knowledgeable enough about the community to say with confidence whether that's true or false.

My personal opinion is a default keymap should be usable for a daily driver right from the start, and I don't think this is, due to the lack of Left Shift and Tab. Perhaps a more conventional keymap should be `default`, and include this one as `(default_)testing` or something similar?

Comment (mechmerlin) [reply keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c]: 🤔

Comment (drashna) [reply keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c]: Maybe have a "dev" keymap for this?

File: keyboards/exclusive/e6v2/bmc/rules.mk (hunk 2 above)
Comment (drashna) [reply keyboards/exclusive/e6v2/bmc/rules.mk]: The reason I cry, is that we have the i2c_master that "should" be used. 

https://github.com/qmk/qmk_firmware/blob/master/drivers/avr/i2c_master.h#L20-L25

You should be able to swap these out, but you'd need to to add a timeout to the functions, and replace the header file includes.

```suggestion
SRC += i2c_master.c
```

File: keyboards/exclusive/e6v2/bmc/bmc.c [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "bmc.h"
+#include "rgblight.h"
+#include "i2c.h"
```
Comment (drashna) [inline keyboards/exclusive/e6v2/bmc/bmc.c]:
```suggestion
#include "i2c_master.h"
```

File: keyboards/exclusive/e6v2/bmc/bmc.c [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    if (!rgblight_config.enable) {
+        for (uint8_t i = 0; i < RGBLED_NUM; i++) {
+            led[i].r = 0;
+            led[i].g = 0;
+            led[i].b = 0;
+        }
+    }
+
+    i2c_init();
+    i2c_send(0xb0, (uint8_t*)led, 3 * RGBLED_NUM);
```
Comment (drashna) [inline keyboards/exclusive/e6v2/bmc/bmc.c]:
```suggestion
    i2c_transmit(0xb0, (uint8_t*)led, 3 * RGBLED_NUM, 100);
```

File: keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c (hunk 1 above)
Comment (mechmerlin) [reply keyboards/exclusive/e6v2/bmc/keymaps/default/keymap.c]: I just put the eep_rst in layer 1

**Comment by drashna** [comment]
Thanks!

**Pull request merged**